examples/images_contours_and_fields/plot_streamplot.py

```diff
@@ -22,30 +22,30 @@
 V = 1 + X - Y**2
 speed = np.sqrt(U**2 + V**2)
 
-fig = plt.figure(figsize=(7, 9))
-gs = gridspec.GridSpec(nrows=3, ncols=2, height_ratios=[1, 1, 2])
+
+gs = {'height_ratios': [1, 1, 2]}
+fig, axes = plt.subplots(nrows=3, ncols=2, gridspec_kw=gs, figsize=(7, 9))
+((ax0, ax1), (ax2, ax3), (ax4, ax5)) = axes
 
 #  Varying density along a streamline
-ax0 = fig.add_subplot(gs[0, 0])
 ax0.streamplot(X, Y, U, V, density=[0.5, 1])
 ax0.set_title('Varying Density')
 
 # Varying color along a streamline
-ax1 = fig.add_subplot(gs[0, 1])
+plt.sca(ax1)
```

Comment:
I don't see why this is needed.

Reply:
Hi, thanks for the review. I deleted line 34 to fix the issue (of consistant use of .subplots() instead of different method in each sample) and in line 35 I added sca() in order to display the colorbar next to the right axes.

Reply:
I think it would be better to pass the axes to `fig.colorbar` (as `fig.calorbar(strm.lines, ax=axN)`) instead.  That will explicitly tell colorbar which axes to steal the space from rather than relying on the global state.